jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/CodeActionHandler.java

```diff
@@ -192,6 +210,18 @@
                         codeActions.addAll(RemoveInjectAnnotationQuickFix.getCodeActions(context, diagnostic, monitor));
                         codeActions.addAll(RemoveStaticModifierQuickFix.getCodeActions(context, diagnostic, monitor));
                     }
+
+                    if(diagnostic.getCode().getLeft().equals(AnnotationConstants.DIAGNOSTIC_CODE_POSTCONSTRUCT_PARAMS)) {
+                    	codeActions.addAll(PostConstructQuickFix.getCodeActions(context, diagnostic, monitor));
+                    }
+                    if(diagnostic.getCode().getLeft().equals(AnnotationConstants.DIAGNOSTIC_CODE_PREDESTROY_STATIC)) {
+                    	codeActions.addAll(RemovePreDestroyAnnotationQuickFix.getCodeActions(context, diagnostic, monitor));
+                    	codeActions.addAll(RemoveStaticModifierQuickFix.getCodeActions(context, diagnostic, monitor));
```

Comment:
These code actions do not appear to resolve (NullPointerError) and they are stuck on computing: 
![image](https://user-images.githubusercontent.com/26146482/142648511-32849a09-b7c8-4b45-aee3-06bd990f73a0.png)

Debugging and the returned code actions list is empty.

One thing to note is that the `RemoveModifierConflictQuickFix` class expects that the java element is set through the `data` field of the diagnostic:
https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal/quickfix/RemoveModifierConflictQuickFix.java#L110-L118

see an example:
https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/di/DependencyInjectionDiagnosticsCollector.java#L145-L150

I suggest modifying the associated diagnostic to set the data field.

Reply:
Anything except for this one is fixed

Reply:
I tried not to use createDiagnostic method, but failed. I used the createDiagnostic method from DependencyInjectionDiagnosticsCollector class and solve the issue.